### PR TITLE
test: update to run matrix tests for all datastore

### PR DESCRIPTION
**Pull request by adriantam**
## Description
Currently the matrix tests are only run for in-memory store.  We want to have it run for all stores. 

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: tests/check/check.go [hunk 1]
```diff
@@ -1368,6 +1368,6 @@ func assertListUsers(ctx context.Context, t *testing.T, assertion *checktest.Ass
 	}
 }
 
-func runTestMatrixSuite(t *testing.T, client ClientInterface) {
+func RunTestMatrixSuite(t *testing.T, client ClientInterface) {
```
Comment (adriantam) [inline tests/check/check.go:1371]: This allows other projects that use OpenFGA as library to be able to run test matrix suite for their project's datastore.

File: tests/check/check_test.go [hunk 2]
```diff
@@ -60,26 +54,42 @@ func testRunTestMatrix(t *testing.T, engine string, experimental bool) {
...
-		runTestMatrixSuite(t, openfgav1.NewOpenFGAServiceClient(conn))
+		RunTestMatrixSuite(t, openfgav1.NewOpenFGAServiceClient(conn))
 	})
 }
 
 func TestCheckMemory(t *testing.T) {
 	testRunAll(t, "memory")
 }
 
+func TestMatrixMemory(t *testing.T) {
```
Comment (elbuo8) [inline tests/check/check_test.go:65]: Why not put it in `TestCheckMemory`?

Comment (adriantam) [reply tests/check/check_test.go:65]: unfortunately the goleak.VerifyNone does not play nicely with the testRunAll :(

**Review by miparnisari — COMMENTED** [review]
What is the reasoning for doing this? An author of a datastore should _not_ have to run these tests to verify whether they wrote it correctly. I realize that this is not the case today, but i would like to not aggravate the problem: https://github.com/openfga/openfga/pull/1615#issuecomment-2272577165

**Pull request closed without merging**